packages/messaging/android/src/main/java/io/invertase/firebase/messaging/ReactNativeFirebaseMessagingStoreImpl.java

```diff
@@ -0,0 +1,52 @@
+package io.invertase.firebase.messaging;
+
+import android.util.Log;
+
+import com.facebook.react.bridge.WritableMap;
+import com.google.firebase.messaging.RemoteMessage;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import io.invertase.firebase.common.UniversalFirebasePreferences;
+
+import static io.invertase.firebase.messaging.JsonConvert.jsonToReact;
+import static io.invertase.firebase.messaging.JsonConvert.reactToJSON;
+import static io.invertase.firebase.messaging.ReactNativeFirebaseMessagingSerializer.remoteMessageFromReadableMap;
+import static io.invertase.firebase.messaging.ReactNativeFirebaseMessagingSerializer.remoteMessageToWritableMap;
+
+public class ReactNativeFirebaseMessagingStoreImpl implements ReactNativeFirebaseMessagingStore {
+
+  @Override
+  public void storeFirebaseMessage(RemoteMessage remoteMessage) {
+    try {
+      String remoteMessageString = reactToJSON(remoteMessageToWritableMap(remoteMessage)).toString();
+      Log.d("storeFirebaseMessage", remoteMessageString);
+      UniversalFirebasePreferences.getSharedInstance().setStringValue(remoteMessage.getMessageId(), remoteMessageString);
```

Comment:
@Salakar - what do you think of this as a design? The goal is to persist notification state across app restarts in case the notification is tapped at some future date, so getInitialNotification works in all cases. So this turns the notification into JSON and stuffs it into preferences for later retrieval.